backend/tournesol/tests/test_text_search.py

```diff
@@ -0,0 +1,712 @@
+            self._make_url(query, "fr"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+
+    def test_stemming(self):
+        """
+        Only the root of words is indexed or queried.
+        This allows to match other words that have the same root,
+        and thus usually the same meaning.
+        """
+        word = "fruitful"
+        query = "fruits"
+        self._create_rated_entity(field1=word)
+
+        response = self.client.get(
+            self._make_url(query),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], self.setup_results_count + 1)
+
+    def test_stemming_french(self):
+        """
+        Each language has its stemming dictionary.
+        Check that it works for french too.
+        """
+        word = "construction"
+        query = "constructif"
+        self._create_rated_entity(field1=word, language="fr")
+
+        response = self.client.get(
+            self._make_url(query, "fr"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_languages_filter(self):
+        """
+        There should be a filter on the entities' languages
+        """
+        self._create_rated_entity(field1=self.field, language="fr")
+        self._create_rated_entity(field1=self.field, language="fr")
+
+        response = self.client.get(self.url, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], self.setup_results_count)
+
+    def test_multiple_languages(self):
+        """
+        If the recommendation filter includes multiple languages,
+        each entity should have a relevance score equal to the
+        max relevance score over all languages.
+        """
+        self._create_rated_entity(field1=self.field, language="en")
+        self._create_rated_entity(field1=self.field, language="fr")
+
+        response = self.client.get(
+            self.url_base + f"?metadata[language]=en&metadata[language]=fr&search={self.query}",
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 2 + self.setup_results_count)
+
+    def test_languages_unknown_by_postgres(self):
+        """
+        Postgres full-text search does not support as many languages
+        as the Tournesol query languages filter.
+
+        If all the languages of the query are not supported by Postgres,
+        a classic case-insensitive, language-independent search is done.
+
+        Just check that the search still works.
+        """
+        word = "kook"
+        self._create_rated_entity(field1=word, language="et")

```

Comment:
Should we use `self.language_unknown_by_postgres` here? and in every place where "et" is used?